klippy/extras/save_variables.py

```diff
@@ -0,0 +1,62 @@
+# Get and Save arbitrary variables in the config file so that values can be kept
+# across restarts.
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import sys,os,logging, ConfigParser
+
+error = ConfigParser.Error
+
+class SaveVariables:
+    error = ConfigParser.Error
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        self.gcode = self.printer.lookup_object('gcode')
+        self.name = config.get_name()
+        self.cfilename = config.get('filename')
+        self.filename = os.path.join(os.path.expanduser("~"), self.cfilename)
+        self.allVariables = {}
+        self.loadVariables()
+        logging.info(str(self.allVariables))
```

Comment:
When logging, it's important to provide context to the log message - so, for example, `logging.info("save_variable state: %s", self.allVariables)`.

Reply:
Done